rust/blockstore/src/arrow/provider.rs

```diff
@@ -252,14 +254,14 @@ impl BlockManager {
         Block::from_record_batch(delta_id, record_batch)
     }
 
-    pub(super) fn cached(&self, id: &Uuid) -> bool {
-        self.block_cache.get(id).is_some()
+    pub(super) async fn cached(&self, id: &Uuid) -> bool {
+        self.block_cache.get(id).await.ok().is_some()
```

Comment:
will this panic in case of errors? Should we instead catch the error and surface it?

Reply:
`.ok().is_some()` should be written `.is_ok()`.  It will not panic.